rdt/transformers/base.py

```diff
@@ -9,20 +9,121 @@ class BaseTransformer:
     and ``fit_transform`` method is already implemented.
     """
 
-    def fit(self, data):
+    INPUT_TYPE = None
+    OUTPUT_TYPES = None
+    DETERMINISTIC_TRANSFORM = None
+    DETERMINISTIC_REVERSE = None
+    COMPOSITION_IS_IDENTITY = None
+    NEXT_TRANSFORMERS = None
+
+    def get_input_type(self):
```

Comment:
Can we make this a static or class method? This will be accessed by the `HyperTransformer` to figure out which transformers to use for a data type

Reply:
Sounds reasonable.